ArduPlane/GCS_Mavlink.cpp

```diff
@@ -791,13 +792,197 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_do_reposition(const mavlink_com
     return MAV_RESULT_FAILED;
 }
 
+// these are GUIDED mode commands that are RATE or slew enabled, so you can have more powerful control than default controls.
+MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_guided_slew_commands(const mavlink_command_int_t &packet)
+{
+
+
+  ::printf("GCS_MAVLINK_Plane::handle_command_int_guided_slew_commands .. slew packet recieved:%d\n",packet.command);
+
+  switch(packet.command) {
+    
+
+#if OFFBOARD_GUIDED == ENABLED
+    case MAV_CMD_GUIDED_CHANGE_SPEED: {
+        // command is only valid in guided
+        if (plane.control_mode != &plane.mode_guided) {
+            gcs().send_text(MAV_SEVERITY_WARNING,"TEMPORARILY_REJECTED - not in guided mode.");
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+         // only airspeed commands are supported right now...
+        if (int(packet.param1) != SPEED_TYPE_AIRSPEED) {  // since SPEED_TYPE is int in range 0-1 and packet.param1 is a *float* this works.
+            ::printf("unsupported packet.param1 in cmd-guided-change-speed:%f\n",packet.param1);
+            return MAV_RESULT_UNSUPPORTED;
+        }
+
+         // reject airspeeds that are outside of the tuning envelope
+        if (packet.param2 > plane.aparm.airspeed_max || packet.param2 < plane.aparm.airspeed_min) {
+            gcs().send_text(MAV_SEVERITY_WARNING,"FAILED - not inside tuning envelope.");
+            return MAV_RESULT_FAILED;
+        }
+
+         // reject duplicate airspeeds
+        float new_target_airspeed_cm = packet.param2 * 100;
+//        if (new_target_airspeed_cm == plane.guided_state.target_airspeed_cm) {
+        if ( fabs(new_target_airspeed_cm - plane.guided_state.target_airspeed_cm)<0.001) { // compare if two floats are 'the same'
```

Comment:
Why do you need to nack this?

Reply:
using both  MAV_RESULT_UNSUPPORTED and MAV_RESULT_FAILED ... it was mostly to allow these two states to be differentiated by the consumer, but its not a big deal as they are both error states.